File: fuzz/corpus/fuzz_objc_parser/seed_simple.m
#import <Foundation/Foundation.h>

@protocol Speakable <NSObject>
- (NSString *)speak;
@end

@interface Greeter : NSObject <Speakable>
@property (nonatomic, copy) NSString *prefix;
- (NSString *)greetName:(NSString *)name loudly:(BOOL)loudly;
@end

@implementation Greeter
- (NSString *)speak {
    return self.prefix;
}

- (NSString *)greetName:(NSString *)name loudly:(BOOL)loudly {
    return [NSString stringWithFormat:@"%@ %@", self.prefix, name];
}
@end

File: fuzz/corpus/fuzz_swift_parser/seed_simple.swift
import Foundation

protocol Greeter {
    func greet(name: String) -> String
}

struct Hello: Greeter {
    let prefix: String

    func greet(name: String) -> String {
        return "\(prefix) \(name)"
    }
}

enum Mode {
    case quiet
    case loud
}
